extension/src/pipeline/quickPick.ts

```diff
@@ -79,26 +142,31 @@ const pickFields = async (
     title: Title.SELECT_PLOT_X_METRIC
   })) as { file: string; key: string }[] | undefined
 
-  if (!xValues || xValues.length === 0) {
+  const xFieldInfo = verifyXFields(xValues)
+
+  if (!xFieldInfo) {
     return
   }
 
+  const { x, firstXKey } = xFieldInfo
+
   const yValues = (await quickPickUserOrderedValues(
-    items.filter(item => xValues.every(val => !isEqual(val, item.value))),
+    items.filter(item => xValues?.every(val => !isEqual(val, item.value))),
     { title: Title.SELECT_PLOT_Y_METRIC }
   )) as { file: string; key: string }[] | undefined
 
-  if (!yValues || yValues.length === 0) {
+  const yFieldInfo = verifyYFields(yValues, xValues?.length || 0)
```

Comment:
`verifyYFields` should have this signature instead: `(yValues: { file: string; key: string }[] | undefined, xFieldsLength = 0)`